spec.html

```diff
@@ -2,20 +2,490 @@
+        <h1>
+          AsyncContextSnapshot (
+          ): a List of Records with fields [[AsyncContextKey]] (a Symbol) and [[AsyncContextvalue]] (an ECMAScript language value)
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It is used to snapshot the surrounding agent's Agent Record's [[AsyncContextMapping]].</dd>
+        </dl>
+        <emu-alg>
+          1. Let _agentRecord_ be the surrounding agent's Agent Record.
+          1. Let _asynccontextMapping_ be _agentRecord_.[[AsyncContextMapping]].
+          1. Let _snapshot_ be a new empty List.
+          1. For each Record { [[AsyncContextKey]], [[AsyncContextValue]] } _p_ of _asynccontextMapping_, do
+            1. If _p_.[[AsyncContextKey]] is not ~empty~, then
+              1. Let _q_ be the Record { [[AsyncContextKey]]: _p_.[[AsyncContextKey]], [[AsyncContextValue]]: _p_.[[AsyncContextValue]] }.
+              1. Append _q_ to _snapshot_.
+          1. Return _snapshot_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-asynccontext-swap" type="abstract operation">
+        <h1>
+          AsyncContextSwap (
+            _snapshot_: a List of Records with fields [[AsyncContextKey]] (a Symbol) and [[AsyncContextvalue]] (an ECMAScript language value)
+          ): a List of Records with fields [[AsyncContextKey]] (a Symbol) and [[AsyncContextvalue]] (an ECMAScript language value)
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It is used to swap the surrounding agent's Agent Record's [[AsyncContextMapping]] with the _snapshot_.</dd>
+        </dl>
+        <emu-alg>
+          1. Let _agentRecord_ be the surrounding agent's Agent Record.
+          1. Let _asynccontextMapping_ be _agentRecord_.[[AsyncContextMapping]].
+          1. Set _agentRecord_.[[AsyncContextMapping]] to _snapshot_.
+          1. Return _asynccontextMapping_.
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-asynccontext-constructor">
+      <h1>The AsyncContext Constructor</h1>
+      <p>The AsyncContext constructor:</p>
+      <ul>
+        <li>is <dfn>%AsyncContext%</dfn>.</li>
+        <li>is the initial value of the *"AsyncContext"* property of the global object.</li>
+        <li>creates and initializes a new AsyncContext when called as a constructor.</li>
+        <li>is not intended to be called as a function and will throw an exception when called in that manner.</li>
+        <li>may be used as the value in an `extends` clause of a class definition. Subclass constructors that intend to inherit the specified AsyncContext behaviour must include a `super` call to the AsyncContext constructor to create and initialize the subclass instance with the internal state necessary to support the `AsyncContext` and `AsyncContext.prototype` built-in methods.</li>
+      </ul>
+
+      <emu-clause id="sec-asynccontext">
+        <h1>AsyncContext ( )</h1>
```

Comment:
I've been thinking about 2 extensions to the constructor:

1. A `name` string
2. A `defaultValue` value

The `name` string is intended to aid debugging in dev-tools. I imagine they might want to display the global context mapping, and it'd be more helpful if we could annotate that list in some way. This could be stored on the symbol's description.

The `defaultValue` would be returned instead of `undefined` whenever the context hasn't been run in this call stack. This is sometimes useful in React.